LibCarla/source/carla/trafficmanager/LocalizationUtils.cpp

```diff
@@ -232,6 +233,43 @@ namespace LocalizationConstants {
     }
 
   }
+  SimpleWaypointPtr TrackTraffic::GetTargetWaypoint(Buffer& waypoint_buffer,const float& target_point_distance,uint64_t& index)
+  {
+    SimpleWaypointPtr target_waypoint = waypoint_buffer.front();
+    uint64_t  startPosn  = static_cast<uint64_t>(std::fabs(target_point_distance/SAMPLING_RESOLUTION));
+    bool mScanForward = false;
+    double target_point_dist_power = std::pow(target_point_distance, 2);
+
+    /// Condition to determine forward or backward scanning of  WayPoint Buffer.
+    if((startPosn < waypoint_buffer.size()) &&
+        (waypoint_buffer.front()->DistanceSquared(target_waypoint)
+        <= target_point_dist_power)) {
+      mScanForward = true;
+    }
+
+    if(mScanForward) {
+      for (uint64_t j = startPosn ;
+          (j < waypoint_buffer.size()) &&
+          (waypoint_buffer.front()->DistanceSquared(target_waypoint)
+          < target_point_dist_power);
+          ++j) {
+          target_waypoint = waypoint_buffer.at(j);
+          index = j;
+          }
+    }
+    else
+    {
+      for (uint64_t j = startPosn ;
+        (j >= 0) &&
+        (waypoint_buffer.front()->DistanceSquared(target_waypoint)
+        < target_point_dist_power);
```

Comment:
I think this should be > target_point_dist_power as the condition for scanning backward is that the distance from the front waypoint to currently predicted target waypoint is more than the target distance (Opposite of the looping condition in the if block above).

Reply:
Agreed and implemented